crates/rpc/rpc/src/eth/revm_utils.rs

```diff
@@ -51,10 +51,48 @@ impl EvmOverrides {
         Self { state, block: None }
     }
 
+    /// Creates a new instance with the given block overrides.
+    pub const fn block(block: Option<Box<BlockOverrides>>) -> Self {
+        Self { state: None, block }
+    }
+
     /// Returns `true` if the overrides contain state overrides.
     pub const fn has_state(&self) -> bool {
         self.state.is_some()
     }
+
+    /// Returns `true` if the overrides contain block overrides.
+    pub const fn has_block(&self) -> bool {
+        self.block.is_some()
+    }
+
+    /// Adds state overrides to an existing instance.
+    pub fn with_state(mut self, state: Option<StateOverride>) -> Self {
+        self.state = state;
+        self
+    }
+
+    /// Adds block overrides to an existing instance.
+    pub fn with_block(mut self, block: Option<Box<BlockOverrides>>) -> Self {
+        self.block = block;
+        self
+    }
+
+    /// Clears the state overrides.
+    pub fn clear_state(&mut self) {
+        self.state = None;
+    }
+
+    /// Clears the block overrides.
+    pub fn clear_block(&mut self) {
+        self.block = None;
+    }
```

Comment:
hmm, not sure these are useful because you could just call take on the fields

Reply:
Yes I just wanted to clear the state override after having an instance of `EvmOverrides` but can just set to None, this is equivalent. Just removed that.